src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -173,82 +176,105 @@ public void InitializeService(ServiceHost serviceHost)
-            if (!ConnectionService.TryFindConnection(executeParams.OwnerUri, out connInfo))
-            {
-                return requestContext.SendError(SR.QueryServiceQueryInvalidOwnerUri);
-            }
-
-            if (connInfo.ConnectionDetails.MultipleActiveResultSets == null || connInfo.ConnectionDetails.MultipleActiveResultSets == false) {
-                // if multipleActive result sets is not allowed, don't specific a connection and make the ownerURI the true owneruri
-                connInfo = null;
-                executeStringParams.OwnerUri = executeParams.OwnerUri;
-            }
-
-            Func<string, Task> queryCreateFailureAction = message => requestContext.SendError(message);
-
-            ResultOnlyContext<SimpleExecuteResult> newContext = new ResultOnlyContext<SimpleExecuteResult>(requestContext);
-
-            // handle sending event back when the query completes
-            Query.QueryAsyncEventHandler queryComplete = async q =>
+            try
             {
-                Query removedQuery;
-                // check to make sure any results were recieved
-                if (q.Batches.Length == 0 || q.Batches[0].ResultSets.Count == 0) 
+                string randomUri = Guid.NewGuid().ToString();
+                ExecuteStringParams executeStringParams = new ExecuteStringParams
                 {
-                    await requestContext.SendError(SR.QueryServiceResultSetHasNoResults);
-                    ActiveQueries.TryRemove(executeStringParams.OwnerUri, out removedQuery);
-                    return;
-                } 
+                    Query = executeParams.QueryString,
+                    // generate guid as the owner uri to make sure every query is unique
+                    OwnerUri = randomUri
+                };
 
-                var rowCount = q.Batches[0].ResultSets[0].RowCount;
-                // check to make sure there is a safe amount of rows to load into memory
-                if (rowCount > Int32.MaxValue) 
+                // get connection
+                ConnectionInfo connInfo;
+                if (!ConnectionService.TryFindConnection(executeParams.OwnerUri, out connInfo))
                 {
-                    await requestContext.SendError(SR.QueryServiceResultSetTooLarge);
-                    ActiveQueries.TryRemove(executeStringParams.OwnerUri, out removedQuery);
+                    await requestContext.SendError(SR.QueryServiceQueryInvalidOwnerUri);
                     return;
                 }
                 
-                SubsetParams subsetRequestParams = new SubsetParams
+                ConnectParams connectParams = new ConnectParams
                 {
-                    OwnerUri = executeStringParams.OwnerUri,
-                    BatchIndex = 0,
-                    ResultSetIndex = 0,
-                    RowsStartIndex = 0,
-                    RowsCount = Convert.ToInt32(rowCount)
+                    OwnerUri = randomUri,
```

Comment:
By connecting with a random URI, doesn't this incur a lot of overhead by kicking off intellisense discovery and whatnot?